EXTI/Core/Src/functions.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * functions.c
+ *
+ *  Created on: Mar 5, 2020
+ *      Author: rodrigocelmer
+ */
+#include "stm32l053xx.h"
+#include "macros.h"
+
+extern int	user_bt_count,	//counter to debounce the button
+			flag_EXTI,		//flag for EXTI interrupt
+			flag_TIM6,		//flag for TIM6 interrupt
+			flag_frequency;	//flag to indicate if we are increasing or decreasing the frequency
+
+void Blink_Red_Led (void)
+{
+	flag_TIM6 = 0;		//clears TIM6 flag
```

Comment:
as you check the flag_TIM6 in the main file, it is a good practice to clear the flag at the same place that you used it.